### DISCUSSION TITLE
Rendering PlotLine with different types on axis

### DISCUSSION THREAD

**Discussion by johannaheuser** [epezent/implot#394] [Q&A]
I am trying to plot a sinus curve using ImPlot where the x-coordinate is represented by a unix timestamp (uint32_t) and the y-coordinate is a floating point value (float).

I ran into the problem that ImPlot only supports homogeneous overloads for PlotLine in terms of x and y. This way I can only plot uint32_t/uint32_t or float/float.

Is there a way or example how to plot coordinates of different types as described above? Is PlotLineG with a custom getter function maybe a way to do this? Any help is much appreciated :)

**Comment by epezent** [comment]
Hey, sorry for the slow response -- yes, I would use `PlotLineG`  in this particular case.
